### PR TITLE
feat: dependency updates for React 18

### PR DESCRIPTION
**Pull request by madpah**
All `dependencies` and `devDependencies` reviewed and either updated or removed based on current `dev/2.x.x.` direction.

Bunch of code changes to support stricter linting.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,7480 @@
...
+minimatch@^3.0.4, minimatch@^3.0.5, minimatch@^3.1.1, minimatch@^3.1.2:
+  version "3.1.2"
+  resolved "https://registry.npmjs.org/minimatch/-/minimatch-3.1.2.tgz#19cd194bfd3e428f049a70817c038d89ab4be35b"
+  integrity sha512-J7p63hRiAjw1NDEww1W7i37+ByIrOWO5XQQAzZ3VOcL0PNybwpfmV/N05zFAzwQ9USyEcX6t3UO+K5aqBQOIHw==
+  dependencies:
+    brace-expansion "^1.1.7"
+
+minimist@>=1.2.2, minimist@^1.2.0, minimist@^1.2.6:
+  version "1.2.8"
+  resolved "https://registry.npmjs.org/minimist/-/minim
```
[the remark this answers is not in the document]
Comment (madpah) [reply yarn.lock:6205]: @sonatype-lift ignore - there is no newer version of React we can upgrade to!

File: package.json [hunk 2]
```diff
@@ -1,53 +1,100 @@
...
+    "@sonatype/js-sona-types": "^1.1.0",
+    "@sonatype/react-shared-components": "^12.10.30",
+    "@types/chrome": "^0.0.218",
+    "@types/firefox-webext-browser": "^109.0.0",
+    "cash-dom": "^8.1.0",
+    "classnames": "^2.3.2",
+    "localforage": "^1.10.0",
+    "node-window-polyfill": "^1.0.2",
+    "packageurl-js": "^1.0.0",
+    "react": "^18.2.0",
```
[the remark this answers is not in the document]
Comment (madpah) [reply package.json:20]: @sonatype-lift ignore this is not fixable!

**Comment by maurycupitt** [comment]
@madpah - This should merge to dev/2.x.x not main - right?

**Comment by madpah** [comment]
> @madpah - This should merge to dev/2.x.x not main - right?

Totally correct. Updated @maurycupitt .

File: src/components/Options/IQServer/IQServerOptionsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  }, [iqServerApplication, iqServerToken, iqServerURL, iqServerUser, currentScanType]);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const setItem = (setter: any, value: string, key: string) => {
+    setter(value);
+    chrome.storage.local.set({[key]: value}, () => {
+      if (chrome.runtime.lastError) {
+        console.error(chrome.runtime.lastError);
+      } else {
+        console.trace('Set value', key, value);
```
[the remark this answers is not in the document]
Comment (madpah) [reply src/components/Options/IQServer/IQServerOptionsPage.tsx]: Is the above now resolved @maurycupitt ?

**Pull request merged**